### PR TITLE
Update to Django 4.2 (LTS)

### PR DESCRIPTION
**Pull request by tkw1536**
This PR updates to Django 4.2 and updates all dependencies to their newest versions where feasible.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,44 +2,45 @@
 name = "membermanagement"
 version = "0.1.0"
 description = "The Django Application for managing Jacobs University Bremen Alumni Members."
-authors = ["Leonhard Kuboschek <leo@legartis.ai>"]
+authors = ["Tom Wiesing <tkw01536@gmail.com>", "Leonhard Kuboschek <leo@legartis.ai>"]
```
Comment (kuboschek) [inline pyproject.toml:5]: This email is outdated, leo@kuboschek.me is more current

File: donations/migrations/0008_alter_donation_target.py
Comment (kuboschek) [inline donations/migrations/0008_alter_donation_target.py:1]: This seems unrelated to the timezone changes.

**Review by kuboschek — APPROVED** [review]
👍

File: pyproject.toml (hunk 1 above)
Comment (tkw1536) [reply pyproject.toml:5]: Ooops forgot about that one

File: donations/migrations/0008_alter_donation_target.py
Comment (tkw1536) [reply donations/migrations/0008_alter_donation_target.py:1]: Indeed, I don't know why the migration didn't get committed before.

File: pyproject.toml (hunk 1 above)
Comment (tkw1536) [reply pyproject.toml]: Fixed in 18e2393010a10fe0269442d8e15dc66641182b43.
